### ISSUE TITLE
L3 Packet Checksum always fuzzed

### ISSUE THREAD

**Issue by jblu42** [jtpereyda/boofuzz#503]
I am trying boofuzz for L3 fuzzing and wrote a little script for fuzzing ICMP, I use RawL3SocketConnection to build an correct IP packet and calculate the checksum for the IP and ICMP packet with s_checksum. I do not want to fuzz the checksum in some tests, but the checksum is always fuzzed, although I specified fuzzable=False.

s_checksum("ipheader", length=2, algorithm="ipv4", fuzzable=False)

Did I specifiy it wrong or is there something wrong with the option fuzzable?

**Comment by SR4ven** [comment]
Are you sure the checksum field is actually fuzzed or does the data the checksum is generated over change?
I just did some testing with the following code and the checksum always stayed the same.
```python
    s_checksum("test", length=2, algorithm="ipv4", fuzzable=False)
    s_static(b"\x4e\xfe\x01\x08\x00\x00\x40\x00", "test") 
```

If the problem persists, could you provide some sample code which I can try to reproduce the issue with please?

**Comment by jblu42** [comment]
No, i am pretty sure it was fuzzed, I looked at the packets in wireshark and the checksum was replaced by fuzzing values not expected changes for the checksum.

After solving the other issue with the wrong bytes sent, I tried it again with boofuzz 0.3.0 and the checksum is not fuzzed any more, using the same script. I could try to go back to the old boofuzz and see if the problem persists, but currently I am happy, if it works in the latest version. If you are interested in it I will try, otherwise I will just continue to work with the latest version.

Thanks for your great work on Boofuzz, this is much appreciated.
